frontend/src/hooks/useStaticValue.ts

```diff
@@ -0,0 +1,24 @@
+import { useRef } from 'react';
+
+/**
+ * Store a static, immutable value that is initialized lazily. An alternative
+ * to useRef, for when the value is expensive to initialize but doesn't need to
+ * be changed.
+ * @param init The function to initialize the static value
+ * @return The initialized value.
+ */
+const useStaticValue = <T>(init: () => T): T => {
+  const ref = useRef<{ initialized: boolean; value: T | undefined }>({
+    initialized: false,
+    value: undefined,
+  });
+
+  if (!ref.current.initialized) {
+    ref.current = { initialized: true, value: init() };
+  }
+
+  // At this point we know `value` is a T, but we  have to convince the typechecker
+  return ref.current.value as T;
```

Comment:
Its probably overkill to use this but did you lookinto a type guard so we dont need the cast https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates

Reply:
Yeah so I thought about that, but because `T` is completely unknown, the caller would have to supply a type guard that can handle type `T`, which would be real annoying. Since we know for 100% sure that `ref.current.value` is a `T` at this point (because we _always_ assign it before hitting this for the first time, and we know for sure that `init()` returns a `T`, I think the cast is safe.